### PR TITLE
Game logic

### PR DESCRIPTION
**Pull request by SevlaMare**
-Initial setup
-Added classes and methods
-Linter fixes
-Review  Improvements

### Game logic
* Player 1 and 2 can choice a spot on board
* Only valid input will be accept, otherwise it will be asked again
* The game end if someone win and winner is announced
* The game end in a draw, if board become full

Observation: On live preview (Reply), 
the import setup is a bit different, due requirements of page.

### CODE REVIEW

File: lib/engine.rb [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+      [0, 3, 6], [1, 4, 7], [2, 5, 8], # j
+      [0, 4, 8], [2, 4, 6] # i, j
+    ]
+
+    mark = (player == 1 ? 'x' : 'o')
+    win.each do |sub_array|
+      return true if sub_array.all? { |element| @field[element] == mark }
+    end
+    false
+  end
```
Comment (kobiyoyo) [inline lib/engine.rb]: This should be another function ,lets implement that.

**Review by kobiyoyo — CHANGES_REQUESTED** [review]
## Hi @SevlaMare       
#### Great Job so far, but there are a few more things required. 
### General
- [x] Add this `.vscode` file to .gitignore
- [x] No inline comments 
Check comments below
### Make changes and submit another code review request.
*Happy coding* 
(Any questions Feel free to contact me on slack @kobiyoyo)

**Review by pelzolga123 — APPROVED** [review]
Approved.

**Comment by pelzolga123** [comment]
## Great work, Congratulation :+1: Project Approved.
You did all changes required previously. 
Please, don't forget to delete the .vscode folder, before you merge this Pull request.
Then go ahead and merge to develop.

_Goodluck in your next milestone_ *Happy coding*

**Pull request merged**